src/rgw/setup/rgw.py

```diff
@@ -545,11 +543,20 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
             current_data_node = socket.gethostname().replace('server', 'data')
             user_status = Rgw._create_admin_on_current_node(conf, current_data_node)
 
-            if user_status != 0:
+            if user_status == 0:
+                Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
+                Rgw._set_consul_kv(rgw_consul_idx, CONSUL_LOCK_KEY, 'user_created')
+            else:
                 machine_ids = Rgw._get_cortx_conf(conf, 'cluster>storage_set[0]>nodes')
-                data_pod_hostnames = [Rgw._get_cortx_conf(conf, \
-                    f'node>{machine_id}>hostname') for machine_id in machine_ids if \
+                data_pod_hostnames = [Rgw._get_cortx_conf(conf,
+                    f'node>{machine_id}>hostname') for machine_id in machine_ids if
                     Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
+                if len(data_pod_hostnames) == 1:
```

Comment:
There should be explicit check if current data node is where last user creation was tried. ideally it would be, but still recommended.
if  len(data_pod_hostnames) == 1 && current_data_node == data_pod_hostnames[0]: